dkist/wcs/models.py

```diff
@@ -739,12 +739,22 @@ def varying_celestial_transform_from_tables(crpix: Union[Iterable[float], u.Quan
 
 
 class Ravel(Model):
-    array_shape = (0, 0)
-    n_inputs = 2
+    n_inputs = None
     n_outputs = 1
     _separable = False
+    _input_units_allow_dimensionless = True
+
+    @property
+    def input_units(self):
+        return {'x': u.pix}
+
+    @property
+    def return_units(self):
+        return {'y': u.pix}
 
     def __init__(self, array_shape, order='C', **kwargs):
+        # n_inputs is a class attribute, but it is unknown until run-time
+        self.__class__.n_inputs = len(array_shape)
```

Comment:
This is a deal breaker. If you had two Ravel models with different dimensions going at the same time this wouldn't work. For example I would expect this to error

```python
m = Ravel((10, 10, 10)) & Ravel((10, 10))
m(0, 1, 2, 3, 4)
```

because you are setting the variable on the Ravel class itself not the instance so both instances of that Ravel model would have `n_inputs = 2` as that would be the last one.

Would this not work if you make it a property? i.e.
```python
@property
def  n_inputs(self):
    return len(self.array_shape)
```

Reply:
Done. This works if you defer the call to super().__init__(**kwargs) until after the instance attributes are set.